modules/videopress/class.videopress-gutenberg.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+class VideoPress_Gutenberg {
+
+	/**
+	 * Initialize the VideoPress Gutenberg extension
+	 */
+	public static function init() {
+		if ( self::should_initialize() ) {
+			add_action( 'init', array( __CLASS__, 'register_video_block_with_videopress' ) );
+		}
+	}
+
+	/**
+	 * Check whether conditions indicate the VideoPress Gutenberg extension should be initialized
+	 *
+	 * @return bool
+	 */
+	public static function should_initialize() {
+		// Should not initialize if Gutenberg is not available
+		if ( ! function_exists( 'register_block_type' ) ) {
+			return false;
+		}
+
+		// Should initialize if this is a WP.com site
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			return true;
+		}
+
+		// Should not initialize if this is a Jetpack site but Jetpack is not active (unless the dev mode is enabled)
+		if ( ! Jetpack::is_active() && ! Jetpack::is_development_mode() ) {
+			return false;
+		}
+
+		// Should initialize by default
+		return true;
+	}
+
+	/**
+	 * Register the Jetpack Gutenberg extension that adds VideoPress support to the core video block.
+	 */
+	public static function register_video_block_with_videopress() {
+		register_block_type( 'core/video', array(
```

Comment:
This may be a stupid question as I did not follow all the conversations about all this, but could we just rely on `Jetpack_Gutenberg` to do all the heavy lifting here?

Instead of using `register_block_type`, could we use `jetpack_register_block`, and that would thus cover all the things we check in `should_initialize` above? See how it's done for the Contact Form block for example:
https://github.com/Automattic/jetpack/blob/60026de776b505d42add576a9eee00a108bd1b5d/modules/contact-form/grunion-contact-form.php#L239

Reply:
See pafL3P-gg-p2 p7jreA-264-p2, the TL;DR is that these VideoPress cases fall outside of the current Jetpack registration assumptions. This patch is intended to be a quicker fix for the user, while we resolve conversations of what we need next. (Either more core contribs or framework investments).

Reply:
we should add a comment here too to warn future devs from coming back around and trying to "fix" this or update to the `jetpack_register_block()` so that they don't start down that path and waste time before realizing it won't work

Reply:
Comment added in 2d9d076